lib/api.ts

```diff
@@ -53,132 +22,159 @@ export interface MemberPublicEditing extends MemberPublic {
-          return [first, last, domain];
-        };
+function focusLookup(member: DocumentData, focuses: DocumentData[]) {
+  const memberFocus = member.fields.focuses;
+  if (memberFocus && Array.isArray(memberFocus) && memberFocus.length !== 0) {
+    return memberFocus.map((foc) => {
+      return (
+        focuses
+          .filter((thisFoc) => foc.id === thisFoc.id)
+          .map((foc) => {
+            return {
+              name:
+                typeof foc.fields["name"] === "string"
+                  ? foc.fields["name"]
+                  : null,
+              id: typeof foc.id === "string" ? foc.id : null,
+            };
+          })[0] || null
+      );
+    });
+  }
+  return null;
+}
 
-        return {
-          name:
-            typeof member.fields["Name"] === "string"
-              ? member.fields["Name"]
-              : null,
-          location:
-            typeof member.fields["Location"] === "string"
-              ? member.fields["Location"]
-              : null,
-          link:
-            typeof member.fields["Link"] === "string"
-              ? member.fields["Link"]
-              : null,
-          title:
-            typeof member.fields["Title"] === "string"
-              ? member.fields["Title"]
-              : null,
-          id:
-            typeof member.fields["RecordID"] === "string"
-              ? member.fields["RecordID"]
-              : null,
-          companySize:
-            typeof member.fields["Company Size"] === "string"
-              ? member.fields["Company Size"]
-              : null,
-          yearsExperience:
-            typeof member.fields["Years of Experience"] === "string"
-              ? member.fields["Years of Experience"]
-              : null,
-          emailAbbr: emailLookup(),
-          region: typeof regionLookup === "string" ? regionLookup : null,
-          industry: industryLookup(),
-          focus: focusLookup(),
-        };
-      })
-      .sort((a, b) => {
-        if (a.name < b.name) {
-          return -1;
-        }
-        if (a.name > b.name) {
-          return 1;
-        }
-        return 0;
-      });
-  });
+function industryLookup(member: DocumentData, industries: DocumentData[]) {
+  const memberIndustry = member.fields.industries;
+  if (
+    memberIndustry &&
+    Array.isArray(memberIndustry) &&
+    memberIndustry.length !== 0
+  ) {
+    return memberIndustry.map((ind) => {
+      return (
+        industries
+          .filter((thisInd) => ind.id === thisInd.id)
+          .map((ind) => {
+            return {
+              name:
+                typeof ind.fields["name"] === "string"
+                  ? ind.fields["name"]
+                  : null,
+              id: typeof ind.id === "string" ? ind.id : null,
+            };
+          })[0] || null
+      );
+    });
+  }
+  return null;
+}
+
+function emailLookup(member: DocumentData) {
+  const memberEmail = member.fields["email"];
```

Comment:
nit: on consistency between using dot (.) and bracket [] notation. Since the key is static, prefer `member.fields.email`.

Reply:
yup, agreed, will change

Reply:
updated in the next commit